app/modules/firebase-db.js

```diff
@@ -54,20 +56,45 @@ export function saveProfileToFb(data) {
       .catch((err) => {
         throw new Error(`Error getting cat profile: ${err}`);
       })
-    );
+  );
 }
 
-export function fetchCatsFromFb(uid) {
+function merge(cats, snaps) {
+  return _.map(cats, (cat, index) => _.assign(
+    {},
+    cat,
+    snaps[index] || {}
+  ));
```

Comment:
Also, I need to confirm what happens when `index` is greater than the length of `snaps`. Initial manual testing shows that it's ok, but when has that ever been reliable?